static/js/subs.js

```diff
@@ -283,6 +283,42 @@ exports.update_settings_for_subscribed = function (sub) {
     stream_edit.show_sub_settings(sub);
 };
 
+exports.update_add_subscriptions_elements = function (allow_user_to_add_subs) {
+    var input_element = $('.add_subscribers_container').find('input[name="principal"]').expectOne();
+    var button_element = $('.add_subscribers_container').find('button[name="add_subscriber"]').expectOne();
+
+    if (allow_user_to_add_subs) {
+        input_element.removeAttr("disabled");
+        button_element.removeAttr("disabled");
+        button_element.css('pointer-events', "");
+        $('.add_subscriber_btn_wrapper').popover('destroy');
+    } else {
+        input_element.attr("disabled", "disabled");
+        button_element.attr("disabled", "disabled");
+
+        // Disabled button blocks mouse events(hover) from reaching
+        // to it's parent div element, so popover don't get triggered.
+        // Add css to prevent this.
+        button_element.css("pointer-events", "none");
+
+        $('.add_subscriber_btn_wrapper').popover({
+            placement: "bottom",
+            content: "<div class='cant_add_subs_hint'>%s</div>".replace(
+                '%s', i18n.t('Only stream subscribers can add users to a private stream.')),
```

Comment:
Not sure, if this is right implementation(`%s`). I tried other solutions I could have thought of, this one seems more appropriate to me out of them.